core/src/main/kotlin/org/evomaster/core/output/TestSuiteOrganizer.kt

```diff
@@ -13,22 +20,131 @@ import org.evomaster.core.search.Solution
  * name each test will have
  */
 class TestSuiteOrganizer {
-
     companion object {
-        fun sortTests(solution: Solution<*>): List<TestCase> {
+        val sortingHelper = SortingHelper()
+        val namingHelper = NamingHelper()
+
+
+        fun sortTests(solution: Solution<*>, customNaming: Boolean = false): List<TestCase> {
 
             //TODO here in the future we will have something bit smarter
-            return naiveSorting(solution)
+            return sortingHelper.sort(solution, namingHelper, customNaming)
+        }
+    }
+}
+
+class NamingHelper {
+    private fun criterion1_500 (individual: EvaluatedIndividual<*>): String{
+        var found500 = false
+        individual.results.forEach {
+            if((it as RestCallResult).getStatusCode() == 500) found500 = true
+        }
+        val suggestedName = when (found500) {
+            true ->  "_with500"
+            false ->  ""
+            else -> throw IllegalStateException("This should not happen")
+        }
+        return suggestedName
+    }
+
+    private fun criterion2_hasPost (individual: EvaluatedIndividual<*>): String{
+        var foundPost = false
+        individual.individual.seeActions().forEach {
+            if ((it as RestCallAction).verb == HttpVerb.POST ) foundPost=true
+        }
+        val suggestedName = when (foundPost) {
+            true ->  "_hasPost"
+            false ->  ""
+            else -> throw IllegalStateException("This should not happen")
+        }
+        return suggestedName
+    }
+
+    private fun criterion3_sampling(individual: EvaluatedIndividual<*>): String{
+        return "_" + (individual.individual as RestIndividual).sampleType
+    }
+
+    private fun criterion4_dbInit(individual: EvaluatedIndividual<*>): String{
+        if ((individual.individual as RestIndividual).dbInitialization.isNotEmpty()){
+            return "_" + "hasDbInit"
+        }
+        else return ""
+    }
+
+    val namingCriteria =  mutableListOf(::criterion1_500, ::criterion2_hasPost, ::criterion3_sampling, ::criterion4_dbInit)
+
+
+    fun suggestName(individual: EvaluatedIndividual<*>): String{
+        var suggestedName = ""
+        namingCriteria.forEach {
+            suggestedName += it(individual)
         }
+        return suggestedName
+    }
+
+}
+
+
+class SortingHelper {
+    /** maxStatusCodeComparatorInd sorts Evaluated individuals based on the highest status code (e.g., 500s are first).
+     * **/
+    val maxStatusCodeComparatorInd = compareBy<EvaluatedIndividual<*>>{ind ->
+        val max = ind.results.maxBy { (it as RestCallResult).getStatusCode()!! }
```

Comment:
again, check all the "as"

Reply:
also, if a sorting only makes sense on RestIndividuals, could have EvaluatedIndividual<RestIndividual> instead of EvaluatedIndividual<*>

Reply:
I kept it as EvaluatedIndividual<*>. In some cases it can be applied on more than just RestIndividuals, so I thought it would be a good idea to keep it more general. 

Having a mix of EvaluatedIndividual<*> and EvaluatedIndividual < RestIndividual > was not possible (compiler was complaining about it).